lib/foreman/command_runner.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'open3'
+
+module Foreman
+  class CommandRunner
+    attr_accessor :command, :stdin, :stdout, :stderr, :status
+
+    def initialize(command_and_args, stdin)
+      self.stdin = stdin
+      self.command = command_and_args
+      # security checks: ensure no shell, absolute path, permissions and executable
+      raise(::Foreman::Exception.new(N_("Command must be array: %s"), command)) unless command.kind_of?(Array)
```

Comment:
Feels like we can ensure this at the settings level

Reply:
We actually do for this use case but this is an utility method that can be used anywhere in the future.